src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvTcpHandle.cs

```diff
@@ -24,20 +25,23 @@ public void Init(UvLoopHandle loop, Action<Action<IntPtr>, IntPtr> queueCloseHan
             _uv.tcp_init(loop, this);
         }
 
-        public void Bind(ServerAddress address)
+        public void Open(long fileDescriptor)
```

Comment:
long/ulong is specced to be exactly 64-bit. I don't think the would be a problem down casting that to a 32-bit IntPtr on 32-bit platforms. Longs are generally a little easier to work with in terms of literals, parsing from strings, etc...

If someone uses a value that was out of range on a 32-bit platform, it wasn't a valid handle to begin with.